oauth.go

```diff
@@ -102,20 +105,22 @@ func GetKMSProvider(configFilePath string) (vaultBaseUrl *string, keyName *strin
 			return nil, nil, nil, err
 		}
 		
-		if (config.ProviderVaultBaseURL != "" && config.ProviderKeyName != "" && config.ProviderKeyVersion != "" ) {
-			vaultBaseUrl = &config.ProviderVaultBaseURL
+		if (config.ProviderVaultName != "" ) {
```

Comment:
what happens if one of these `required` values is missing?

Reply:
If `ProviderVaultName` or `ProviderKeyName` is missing from the config, then we return an error `failed to find KMS providerKeyName in config`. acs-engine should be providing these parameters in `azure.json`.

Reply:
yeah following last comment this now makes sense